openstudiocore/src/model/ElectricLoadCenterInverterLookUpTable.cpp

```diff
@@ -0,0 +1,494 @@
+  }
+
+  boost::optional<double> ElectricLoadCenterInverterLookUpTable_Impl::efficiencyAt100PowerAndNominalVoltage() const {
+    return getDouble(OS_ElectricLoadCenter_Inverter_LookUpTableFields::Efficiencyat100_PowerandNominalVoltage,true);
+  }
+
+  bool ElectricLoadCenterInverterLookUpTable_Impl::setAvailabilitySchedule(Schedule& schedule) {
+    bool result = setSchedule(OS_ElectricLoadCenter_Inverter_LookUpTableFields::AvailabilityScheduleName,
+                              "ElectricLoadCenterInverterLookUpTable",
+                              "Availability",
+                              schedule);
+    return result;
+  }
+
+  void ElectricLoadCenterInverterLookUpTable_Impl::resetAvailabilitySchedule() {
+    bool result = setString(OS_ElectricLoadCenter_Inverter_LookUpTableFields::AvailabilityScheduleName, "");
+    OS_ASSERT(result);
+  }
+
+  bool ElectricLoadCenterInverterLookUpTable_Impl::setThermalZone(const boost::optional<ThermalZone>& thermalZone) {
+    bool result(false);
+    if (thermalZone) {
+      result = setPointer(OS_ElectricLoadCenter_Inverter_LookUpTableFields::ZoneName, thermalZone.get().handle());
+    }
+    else {
+      resetThermalZone();
+      result = true;
+    }
+    return result;
+  }
+
+  void ElectricLoadCenterInverterLookUpTable_Impl::resetThermalZone() {
+    bool result = setString(OS_ElectricLoadCenter_Inverter_LookUpTableFields::ZoneName, "");
+    OS_ASSERT(result);
+  }
+
+  bool ElectricLoadCenterInverterLookUpTable_Impl::setRadiativeFraction(boost::optional<double> radiativeFraction) {
+    bool result(false);
+    if (radiativeFraction) {
+      result = setDouble(OS_ElectricLoadCenter_Inverter_LookUpTableFields::RadiativeFraction, radiativeFraction.get());
+    }
+    else {
+      resetRadiativeFraction();
+      result = true;
+    }
+    return result;
+  }
+
+  void ElectricLoadCenterInverterLookUpTable_Impl::resetRadiativeFraction() {
+    bool result = setString(OS_ElectricLoadCenter_Inverter_LookUpTableFields::RadiativeFraction, "");
+    OS_ASSERT(result);
+  }
+
+  void ElectricLoadCenterInverterLookUpTable_Impl::setRatedMaximumContinuousOutputPower(boost::optional<double> ratedMaximumContinuousOutputPower) {
+    bool result(false);
+    if (ratedMaximumContinuousOutputPower) {
+      result = setDouble(OS_ElectricLoadCenter_Inverter_LookUpTableFields::RatedMaximumContinuousOutputPower, ratedMaximumContinuousOutputPower.get());
+    }
+    else {
+      resetRatedMaximumContinuousOutputPower();
+      result = true;
+    }
+    OS_ASSERT(result);
+  }
+
+  void ElectricLoadCenterInverterLookUpTable_Impl::resetRatedMaximumContinuousOutputPower() {
+    bool result = setString(OS_ElectricLoadCenter_Inverter_LookUpTableFields::RatedMaximumContinuousOutputPower, "");
+    OS_ASSERT(result);
+  }
+
+  void ElectricLoadCenterInverterLookUpTable_Impl::setNightTareLossPower(boost::optional<double> nightTareLossPower) {
```

Comment:
@evanweaver all of these impl functions that take optionals instead of real types are just leftovers from the old qattributes, I typically convert these to not take optionals, the optional signature is redundant with the reset method.  All that said, what really matters is the public interface so I won't be too picky about impls.